anybutton error

Fresh instalation 3.9.16 and same error like on 3.9.15

![Image](https://github.com/user-attachments/assets/d40eb0c7-85dc-424b-b52a-17c37d151d43)

i run  W11 24H2, 

this is my first comment ever on github, so i dont know how this is usually done. but anywho..
I get the first sign that something  isnt right and loads of warnings a this build stage. 

```shell
C:\Users\Admin\displaycal-py3>python -m build
* Creating isolated environment: venv+pip...
* Installing packages in isolated environment:
  - pywin32; platform_system=="Windows"
  - setuptools
* Getting build dependencies for sdist...
<string>:20: DeprecationWarning: 'msilib' is deprecated and slated for removal in Python 3.13
Trying to get git version information...
Trying to get git information...
Generating __version__.py
Version 3.9.16
['egg_info']
*** C:\Users\Admin\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyproject_hooks\_in_process\_in_process.py egg_info
using distutils
C:\Users\Admin\AppData\Local\Temp\build-env-9hrggciv\Lib\site-packages\setuptools\config\pyprojecttoml.py:72: _ExperimentalConfiguration: `[tool.distutils]` in `pyproject.toml` is still *experimental* and likely to change in future releases.
  config = read_configuration(filepath, True, ignore_option_errors, dist)
C:\Users\Admin\AppData\Local\Temp\build-env-9hrggciv\Lib\site-packages\setuptools\config\_apply_pyprojecttoml.py:82: SetuptoolsWarning: `license` overwritten by `pyproject.toml`
  corresp(dist, value, root_dir)
warning: no files found matching 'MANIFEST'
warning: no files found matching 'use-distutils'
warning: no files found matching '_in_process.py'
warning: no files found matching '_in_process.cfg'
warning: no files found matching 'dist\copyright'
warning: no files found matching 'dist\net.displaycal.DisplayCAL.appdata.xml'
warning: no files found matching 'DisplayCAL\quirk.json'
warning: no previously-included files matching '.svn\*' found anywhere in distribution
warning: no previously-included files matching '*~' found anywhere in distribution
warning: no previously-included files matching '*.backup' found anywhere in distribution
warning: no previously-included files matching '*\__pycache__\*' found anywhere in distribution
warning: no previously-included files matching '*.bak' found anywhere in distribution
* Building sdist...
<string>:20: DeprecationWarning: 'msilib' is deprecated and slated for removal in Python 3.13
Trying to get git version information...
Trying to get git information...
Generating __version__.py
Version 3.9.16
['sdist', '--formats', 'gztar', '--dist-dir', 'C:\\Users\\Admin\\displaycal-py3\\dist\\.tmp-bgpsmhkk']
*** C:\Users\Admin\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyproject_hooks\_in_process\_in_process.py sdist --formats gztar --dist-dir C:\Users\Admin\displaycal-py3\dist\.tmp-bgpsmhkk
using distutils
C:\Users\Admin\AppData\Local\Temp\build-env-9hrggciv\Lib\site-packages\setuptools\config\pyprojecttoml.py:72: _ExperimentalConfiguration: `[tool.distutils]` in `pyproject.toml` is still *experimental* and likely to change in future releases.
  config = read_configuration(filepath, True, ignore_option_errors, dist)
C:\Users\Admin\AppData\Local\Temp\build-env-9hrggciv\Lib\site-packages\setuptools\config\_apply_pyprojecttoml.py:82: SetuptoolsWarning: `license` overwritten by `pyproject.toml`
  corresp(dist, value, root_dir)
warning: no files found matching 'MANIFEST'
warning: no files found matching 'use-distutils'
warning: no files found matching '_in_process.py'
warning: no files found matching '_in_process.cfg'
warning: no files found matching 'DisplayCAL\quirk.json'
warning: no previously-included files matching '.svn\*' found anywhere in distribution
warning: no previously-included files matching '*~' found anywhere in distribution
warning: no previously-included files matching '*.backup' found anywhere in distribution
warning: no previously-included files matching '*\__pycache__\*' found anywhere in distribution
warning: no previously-included files matching '*.bak' found anywhere in distribution
* Building wheel from sdist
* Creating isolated environment: venv+pip...
* Installing packages in isolated environment:
  - pywin32; platform_system=="Windows"
  - setuptools
* Getting build dependencies for wheel...
<string>:20: DeprecationWarning: 'msilib' is deprecated and slated for removal in Python 3.13
['egg_info']
*** C:\Users\Admin\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyproject_hooks\_in_process\_in_process.py egg_info
using distutils
C:\Users\Admin\AppData\Local\Temp\build-env-vl98dy_1\Lib\site-packages\setuptools\config\pyprojecttoml.py:72: _ExperimentalConfiguration: `[tool.distutils]` in `pyproject.toml` is still *experimental* and likely to change in future releases.
  config = read_configuration(filepath, True, ignore_option_errors, dist)
C:\Users\Admin\AppData\Local\Temp\build-env-vl98dy_1\Lib\site-packages\setuptools\config\_apply_pyprojecttoml.py:82: SetuptoolsWarning: `license` overwritten by `pyproject.toml`
  corresp(dist, value, root_dir)
warning: no files found matching 'MANIFEST'
warning: no files found matching 'use-distutils'
warning: no files found matching '_in_process.py'
warning: no files found matching '_in_process.cfg'
warning: no files found matching 'DisplayCAL\quirk.json'
warning: no previously-included files found matching 'misc\Argyll'
warning: no previously-included files found matching 'misc\*.rules'
warning: no previously-included files found matching 'misc\*.usermap'
warning: no previously-included files matching '.svn\*' found anywhere in distribution
warning: no previously-included files matching '*~' found anywhere in distribution
warning: no previously-included files matching '*.backup' found anywhere in distribution
warning: no previously-included files matching '*\__pycache__\*' found anywhere in distribution
warning: no previously-included files matching '*.bak' found anywhere in distribution
* Building wheel...
<string>:20: DeprecationWarning: 'msilib' is deprecated and slated for removal in Python 3.13
['bdist_wheel', '--dist-dir', 'C:\\Users\\Admin\\displaycal-py3\\dist\\.tmp-uobrgkw1']
*** C:\Users\Admin\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyproject_hooks\_in_process\_in_process.py bdist_wheel --dist-dir C:\Users\Admin\displaycal-py3\dist\.tmp-uobrgkw1
using distutils
C:\Users\Admin\AppData\Local\Temp\build-env-vl98dy_1\Lib\site-packages\setuptools\config\pyprojecttoml.py:72: _ExperimentalConfiguration: `[tool.distutils]` in `pyproject.toml` is still *experimental* and likely to change in future releases.
  config = read_configuration(filepath, True, ignore_option_errors, dist)
C:\Users\Admin\AppData\Local\Temp\build-env-vl98dy_1\Lib\site-packages\setuptools\config\_apply_pyprojecttoml.py:82: SetuptoolsWarning: `license` overwritten by `pyproject.toml`
  corresp(dist, value, root_dir)
warning: no files found matching 'MANIFEST'
warning: no files found matching 'use-distutils'
warning: no files found matching '_in_process.py'
warning: no files found matching '_in_process.cfg'
warning: no files found matching 'DisplayCAL\quirk.json'
warning: no previously-included files found matching 'misc\Argyll'
warning: no previously-included files found matching 'misc\*.rules'
warning: no previously-included files found matching 'misc\*.usermap'
warning: no previously-included files matching '.svn\*' found anywhere in distribution
warning: no previously-included files matching '*~' found anywhere in distribution
warning: no previously-included files matching '*.backup' found anywhere in distribution
warning: no previously-included files matching '*\__pycache__\*' found anywhere in distribution
warning: no previously-included files matching '*.bak' found anywhere in distribution
Successfully built displaycal-3.9.16.tar.gz and displaycal-3.9.16-py3-none-any.whl
```

does that make any sense? it might be my previous install attempts that is the culprit here, I did the rest of the install anyway and this is what i got:

```shell
C:\Users\Admin\displaycal-py3>pip install dist/displaycal-3.9.16-py3-none-any.whl
Processing c:\users\admin\displaycal-py3\dist\displaycal-3.9.16-py3-none-any.whl
Requirement already satisfied: build in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (1.2.2.post1)
Requirement already satisfied: certifi in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (2024.12.14)
Requirement already satisfied: comtypes in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (1.4.10)
Requirement already satisfied: distro in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (1.9.0)
Requirement already satisfied: numpy in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (2.2.5)
Requirement already satisfied: Pillow in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (11.2.1)
Requirement already satisfied: protobuf in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (4.25.6)
Requirement already satisfied: PyChromecast in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (14.0.7)
Requirement already satisfied: pywin32 in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (310)
Requirement already satisfied: Send2Trash in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (1.8.3)
Requirement already satisfied: WMI in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (1.5.1)
Requirement already satisfied: wxPython>=4.2.2 in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (4.2.3)
Requirement already satisfied: zeroconf in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from displaycal==3.9.16) (0.146.5)
Requirement already satisfied: packaging>=19.1 in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from build->displaycal==3.9.16) (25.0)
Requirement already satisfied: pyproject_hooks in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from build->displaycal==3.9.16) (1.2.0)
Requirement already satisfied: colorama in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from build->displaycal==3.9.16) (0.4.6)
Requirement already satisfied: casttube>=0.2.0 in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from PyChromecast->displaycal==3.9.16) (0.2.1)
Requirement already satisfied: requests in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from casttube>=0.2.0->PyChromecast->displaycal==3.9.16) (2.32.3)
Requirement already satisfied: ifaddr>=0.1.7 in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from zeroconf->displaycal==3.9.16) (0.2.0)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from requests->casttube>=0.2.0->PyChromecast->displaycal==3.9.16) (3.4.1)
Requirement already satisfied: idna<4,>=2.5 in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from requests->casttube>=0.2.0->PyChromecast->displaycal==3.9.16) (3.7)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\users\admin\appdata\local\programs\python\python311\lib\site-packages (from requests->casttube>=0.2.0->PyChromecast->displaycal==3.9.16) (2.4.0)
Installing collected packages: displaycal
Successfully installed displaycal-3.9.16
```

when i try to run it, the wrong windows version is written out here:


```shell
C:\Users\Admin\displaycal-py3>python -m DisplayCAL

> Acquired lock file: <DisplayCAL.main.AppLock object at 0x0000000000FAB490>
> __main__.py 3.9.16 2025-05-09T08:55:21Z
> Windows 10 Pro Version 2009 Build 26100.3775 x86_64

Python 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)]
CA file C:\Users\Admin\AppData\Local\Programs\Python\Python311\Lib\site-packages\certifi\cacert.pem
[Errno 2] No such file or directory: 'C:\\Users\\Admin\\AppData\\Roaming\\DisplayCAL\\logs\\__main__-fault.log'
wxPython 4.2.3 msw (phoenix) wxWidgets 3.2.7
Encoding: utf-8
File system encoding: utf-8
listening
writing to lock file: port: 15411
No module named 'comtypes.gen'
Starting up...
Audio module: pyglet 2.1.6
Initializing GUI...

Handled TypeError in SetBitmapLabel!
self._SetBitmapLabel: <built-in function SetBitmapLabel>
bitmap: <wx._core.Bitmap object at 0x0000000034394E50>
NoneType: None
┌──────────────────────────────────────────────────────────────────────────────┐
│ Traceback (most recent call last):                                           │
│   File "C:\Users\Admin\displaycal-py3\DisplayCAL\wxfixes.py", line 995, in   │
│ set_bitmap_labels                                                            │
│     btn.SetBitmapDisabled(get_bitmap_disabled(bitmap))                       │
│ wx._core.wxAssertionError: C++ assertion "which == State_Normal" failed at   │
│ ..\..\src\msw\anybutton.cpp(819) in wxAnyButton::DoSetBitmap(): Must set     │
│ normal bitmap with the new size first                                        │
│                                                                              │
│ During handling of the above exception, another exception occurred:          │
│                                                                              │
│ Traceback (most recent call last):                                           │
│   File "C:\Users\Admin\displaycal-py3\DisplayCAL\xh_filebrowsebutton.py",    │
│ line 28, in DoCreateResource                                                 │
│     w = self._class(                                                         │
│         ^^^^^^^^^^^^                                                         │
│   File "C:\Users\Admin\displaycal-py3\DisplayCAL\wxwindows.py", line 3184,   │
│ in __init__                                                                  │
│     filebrowse.FileBrowseButton.__init__(self, *arguments, **namedarguments) │
│   File                                                                       │
│ "C:\Users\Admin\AppData\Local\Programs\Python\Python311\Lib\site-packages\wx\lib\filebrowsebutton.py", │
│ line 76, in __init__                                                         │
│     self.createDialog(parent, id, pos, size, style, name )                   │
│   File "C:\Users\Admin\displaycal-py3\DisplayCAL\wxwindows.py", line 3333,   │
│ in createDialog                                                              │
│     self.browseButton = self.createBrowseButton()                            │
│                         ^^^^^^^^^^^^^^^^^^^^^^^^^                            │
│   File "C:\Users\Admin\displaycal-py3\DisplayCAL\wxwindows.py", line 3309,   │
│ in createBrowseButton                                                        │
│     button = GenBitmapButton(                                                │
│              ^^^^^^^^^^^^^^^^                                                │
│   File "C:\Users\Admin\displaycal-py3\DisplayCAL\wxfixes.py", line 1210, in  │
│ __init__                                                                     │
│     set_bitmap_labels(self)                                                  │
│   File "C:\Users\Admin\displaycal-py3\DisplayCAL\wxfixes.py", line 999, in   │
│ set_bitmap_labels                                                            │
│     btn.SetBitmapDisabled(get_bitmap_disabled(bitmap))                       │
│ wx._core.wxAssertionError: C++ assertion "which == State_Normal" failed at   │
│ ..\..\src\msw\anybutton.cpp(819) in wxAnyButton::DoSetBitmap(): Must set     │
│ normal bitmap with the new size first                                        │
```

--- a/DisplayCAL/wxfixes.py
+++ b/DisplayCAL/wxfixes.py
@@ -991,7 +991,12 @@ def set_bitmap_labels(btn, disabled=True, focus=None, pressed=True):
 
     # Disabled
     if disabled:
-        btn.SetBitmapDisabled(get_bitmap_disabled(bitmap))
+        try:
+            btn.SetBitmapDisabled(get_bitmap_disabled(bitmap))
+        except wx._core.wxAssertionError:
+            # must set normal bitmap first, fixes #507
+            btn.SetBitmapLabel(bitmap)
+            btn.SetBitmapDisabled(get_bitmap_disabled(bitmap))
 
     # Focus/Hover
     if sys.platform != "darwin" and focus is not False:

--- a/DisplayCAL/wxfixes.py
+++ b/DisplayCAL/wxfixes.py
@@ -474,12 +474,7 @@ wx.AnyButton._SetBitmapLabel = wx.AnyButton.SetBitmapLabel
 def SetBitmapLabel(self, bitmap):
     """Override the SetBitmapLabel to avoid flickering."""
     if self.GetBitmapLabel() != bitmap:
-        try:
-            self._SetBitmapLabel(bitmap)
-        except TypeError:
-            print("Handled TypeError in SetBitmapLabel!")
-            print(f"self._SetBitmapLabel: {self._SetBitmapLabel}")
-            print(f"bitmap: {bitmap}")
+        self._SetBitmapLabel(self, bitmap)
 
 wx.AnyButton.SetBitmapLabel = SetBitmapLabel
 
@@ -991,12 +986,11 @@ def set_bitmap_labels(btn, disabled=True, focus=None, pressed=True):
 
     # Disabled
     if disabled:
+        disabled_bitmap = get_bitmap_disabled(bitmap)
         try:
-            btn.SetBitmapDisabled(get_bitmap_disabled(bitmap))
+            btn.SetBitmapDisabled(disabled_bitmap)
         except wx._core.wxAssertionError:
-            # must set normal bitmap first, fixes #507
-            btn.SetBitmapLabel(bitmap)
-            btn.SetBitmapDisabled(get_bitmap_disabled(bitmap))
+            pass
 
     # Focus/Hover
     if sys.platform != "darwin" and focus is not False:
@@ -1007,10 +1001,16 @@ def set_bitmap_labels(btn, disabled=True, focus=None, pressed=True):
         else:
             # Use focus bitmap
             bmp = get_bitmap_focus(bitmap)
-        btn.SetBitmapFocus(bmp)
+        try:
+            btn.SetBitmapFocus(bmp)
+        except wx._core.wxAssertionError:
+            pass
         if hasattr(btn, "SetBitmapCurrent"):
             # Phoenix
-            btn.SetBitmapCurrent(bmp)
+            try:
+                btn.SetBitmapCurrent(bmp)
+            except wx._core.wxAssertionError:
+               pass
         else:
             # Classic
             btn.SetBitmapHover(bmp)
@@ -1023,7 +1023,10 @@ def set_bitmap_labels(btn, disabled=True, focus=None, pressed=True):
             bmp = get_bitmap_pressed(bitmap)
         if hasattr(btn, "SetBitmapPressed"):
             # Phoenix
-            btn.SetBitmapPressed(bmp)
+            try:
+                btn.SetBitmapPressed(bmp)
+            except wx._core.wxAssertionError:
+                pass
         else:
             # Classic
             btn.SetBitmapSelected(bmp)

--- a/DisplayCAL/wxfixes.py
+++ b/DisplayCAL/wxfixes.py
@@ -1,6 +1,7 @@
 # -*- coding: utf-8 -*-
 
 import os
+import platform
 import re
 import shutil
 import sys
@@ -468,16 +469,19 @@ else:
     wx_Panel = wx.Panel
 
 
-wx.BitmapButton._SetBitmapLabel = wx.BitmapButton.SetBitmapLabel
-
+wx.AnyButton._SetBitmapLabel = wx.AnyButton.SetBitmapLabel
 
 def SetBitmapLabel(self, bitmap):
-    """Replacement for SetBitmapLabel which avoids flickering"""
+    """Override the SetBitmapLabel to avoid flickering."""
     if self.GetBitmapLabel() != bitmap:
-        self._SetBitmapLabel(bitmap)
+        try:
+            self._SetBitmapLabel(bitmap)
+        except TypeError:
+            print("Handled TypeError in SetBitmapLabel!")
+            print(f"self._SetBitmapLabel: {self._SetBitmapLabel}")
+            print(f"bitmap: {bitmap}")
 
-
-wx.BitmapButton.SetBitmapLabel = SetBitmapLabel
+wx.AnyButton.SetBitmapLabel = SetBitmapLabel
 
 
 def BitmapButtonEnable(self, enable=True):
@@ -982,15 +986,12 @@ def set_bitmap_labels(btn, disabled=True, focus=None, pressed=True):
         if -1 in size:
             size = (16, 16)
         bitmap = wx.ArtProvider.GetBitmap(wx.ART_MISSING_IMAGE, size=size)
+        # Ensure the normal bitmap is set first, fixes #507
+        btn.SetBitmapLabel(bitmap)
 
     # Disabled
     if disabled:
-        try:
-            btn.SetBitmapDisabled(get_bitmap_disabled(bitmap))
-        except wx._core.wxAssertionError:
-            # must set normal bitmap first, fixes #507
-            btn.SetBitmapLabel(bitmap)
-            btn.SetBitmapDisabled(get_bitmap_disabled(bitmap))
+        btn.SetBitmapDisabled(get_bitmap_disabled(bitmap))
 
     # Focus/Hover
     if sys.platform != "darwin" and focus is not False:

--- a/misc/build_and_launch.cmd
+++ b/misc/build_and_launch.cmd
@@ -1,5 +1,5 @@
 rmdir /s /q dist
 %LOCALAPPDATA%\Programs\Python\Python311\python.exe -m pip uninstall -y displaycal
 %LOCALAPPDATA%\Programs\Python\Python311\python.exe -m build
-%LOCALAPPDATA%\Programs\Python\Python311\python.exe -m pip install --upgrade dist\displaycal-3.9.15-py3-none-any.whl
+%LOCALAPPDATA%\Programs\Python\Python311\python.exe -m pip install --upgrade dist\displaycal-3.9.16-py3-none-any.whl
 %LOCALAPPDATA%\Programs\Python\Python311\python.exe -m DisplayCAL

--- a/misc/build_windows_installer.cmd
+++ b/misc/build_windows_installer.cmd
@@ -1,7 +1,7 @@
 rmdir /s /q dist
 %LOCALAPPDATA%\Programs\Python\Python311\python.exe -m pip uninstall -y displaycal
 %LOCALAPPDATA%\Programs\Python\Python311\python.exe -m build
-%LOCALAPPDATA%\Programs\Python\Python311\python.exe -m pip install --upgrade dist\displaycal-3.9.15-py3-none-any.whl
+%LOCALAPPDATA%\Programs\Python\Python311\python.exe -m pip install --upgrade dist\displaycal-3.9.16-py3-none-any.whl
 %LOCALAPPDATA%\Programs\Python\Python311\python.exe DisplayCAL\freeze.py
 %LOCALAPPDATA%\Programs\Python\Python311\python.exe setup.py inno
 cd dist

